ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py

```diff
@@ -7321,3 +7321,34 @@ def test_torch_instance_log10(
         frontend=frontend,
         on_device=on_device,
     )
+
+
+# sign
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="torch.tensor",
+    method_name="sign",
+    dtype_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
```

Comment:
Hi @ZiadAmerr, Thank you for the review. sign() only accepts real values. So, it will need to accept complex nos. for valid dtype and for complex nos. PyTorch has a separate API -sgn(), which is an extension of sign(). 

https://pytorch.org/docs/stable/generated/torch.sgn.html

Reply:
So, I converted the test dtype to **numeric** (as the PyTorch frontend of sign API is tested using **numeric**) and mentioned **complex** in the unsupported dtype decorator. The tests pass now.

Reply:
Well you do got a point in following the same structure as before, but we are actually cleaning up the tests now, as you could see from your previous PR, I also stated that we should use `valid` even though the rest of the testing functions used other values. If the sign does not support bool, please state that in the decorator alongside complex. Using valid will generate all dtypes that are not in the decorators. `numeric` does not capture all possible dtypes gor that particular frontend function.

Reply:
Oh okay. Makes sense. Thanks. Will make the changes.